File: Por traducir/ModularInverse.py

def extendedEuclid(a,b):
    if (b == 0):
        x = 1;
        y = 0;
        return a
    
    d = extendedEuclid(b, a % b)
    aux = x
    x = y
    y = aux - ((a / b) * y)
    return d

def modInverse1(a, mod):
    d = extendedEuclid(a, mod)
    if (d > 1):
        return -1;
    return (x % mod + mod) % mod;

def modInverse(mod):
    inv=[0 in range (mod)]
    inv[1] = 1;
    for i in range(2,mod):
        inv[i] = (mod - (mod / i) * inv[mod % i] % mod) % mod
x=0 
y=0

print(modInverse1(3, 7))

File: Por traducir/ModularInverse.cpp
#include <bits/stdc++.h> 
using namespace std;
int x, y;

static int extendedEuclid(int a, int b) {
    if (b == 0) {
        x = 1;
        y = 0;
        return a;
    }
    int d = extendedEuclid(b, a % b);
    int aux = x;
    x = y;
    y = aux - ((a / b) * y);
    return d;
}

long modInverse(int a, int mod) {
    long d = extendedEuclid(a, mod);
    if (d > 1) {
        return -1;
    }
    return (x % mod + mod) % mod;
}

void modInverse(int mod) {
    int inv[mod];
    inv[1] = 1;
    for (int i = 2; i < mod; i++) {
        inv[i] = (mod - (mod / i) * inv[mod % i] % mod) % mod;
    }
}

int main() {
    cout << modInverse(3, 7) << endl;
}

